### ISSUE TITLE
[Pagination]: pagination elements are not grouped into a list

### ISSUE THREAD

**Issue by NatalliaAlieva** [epam/UUI#2717]
## Description
Pagination elements aren't semantically grouped into a list. 

## Steps to Reproduce
1. Open [Paginator](https://uui.epam.com/documents?id=paginator&mode=doc&category=components) component.
2. Open DevTools and inspect the navigation elements in the pagination.

## Actual result
Pagination elements are not grouped into a list.
<img width="1358" alt="Screenshot 2025-01-07 at 17 25 45" src="https://github.com/user-attachments/assets/d586639f-0bb0-43a8-9e1e-9bd3fd1aa5db" />

## Expected result
Pagination elements should be grouped into a semantic list (`<ul> or <ol>`), with each element wrapped in `<li>` tags, to ensure proper navigation and accessibility compliance.
<img width="1361" alt="Screenshot 2025-01-07 at 17 27 12" src="https://github.com/user-attachments/assets/d91d0092-f715-49e0-8a3e-6c4588d255b5" />

**Failed WCAG Checkpoint:** 1.3.1 Info and Relationships (Level A)
https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html

**Comment by NatalliaAlieva** [comment]
A few screenshots as references for how pagination is styled as a list on other websites:
<img width="1338" alt="Screenshot 2025-01-08 at 13 43 16" src="https://github.com/user-attachments/assets/e15c9334-d553-4174-b4c9-6637eb262121" />
<img width="1313" alt="Screenshot 2025-01-08 at 13 56 08" src="https://github.com/user-attachments/assets/07b5cea6-3956-4d51-a78f-106f414b9bf7" />
